### PR TITLE
Prepare 2.5.0-alpha.1 release

### PR DESCRIPTION
**Pull request by vaszig**
DONE ~~Needs rebasing after #739 has been approved and merged.~~

Closes open-formulieren/open-forms#4849 (partially)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,79 @@
...
+--------
+
+* [#4827] Do not add initial_data_reference if empty.
+* [#4546] Properly pass initial_data_reference to the nextUrl query params on the login URL.
+
+Project maintenance
+-------------------
+
+* Ensure prettier checks jsx files.
+* [#4849] Ensure .jsx files are picked up for translation as well.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:61]:
```suggestion
* Ensured prettier checks jsx files.
* [#4849] Ensured .jsx files are picked up for translation as well.
```
we use past tense for the changelogs

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,79 @@
...
+* [#3283] Removed password Formio component.
+* Fixed button stories/documentation page.
+* [#4320] Updated stories and UX of completion view.
+* Excluded file renames from git blame.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:65]:
```suggestion
```
kind of implied by the JSX renames, it just has to be a separate commit otherwise `git-blame-ignore-revs` doesn't work properly.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,79 @@
...
+* Renamed .js files to .jsx.
+* Removed old tilde prefix in SCSS imports.
+* [#4546] Refactored aria attributes utilities.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:68]: That's not something from that ticket, is it? I think one of the a11y stories, but you can also omit it from the changelog.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -2,6 +2,79 @@
...
+* Upgraded to Jest 29 because of MSW requirements.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:69]: You *could* mention #724 :)

File: CHANGELOG.rst [hunk 5]
```diff
@@ -2,6 +2,79 @@
...
+* Updated story after translations update.
+* Updated developer docs about mocking in stories.
+* Handled msw-storybook-addon breaking change.
+* Updated API mocks following breaking changes in MSW.
+* Upgraded to MSW 2.5.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:74]:
```suggestion
* Upgraded to MSW 2.5.
```
all the rest is related to/because of this

File: CHANGELOG.rst [hunk 6]
```diff
@@ -2,6 +2,79 @@
...
+* Updated actions with explicit mocks.
+* Upgraded to Storybook 8.3.6 with upgrade script.
```
Comment (sergei-maertens) [inline CHANGELOG.rst:76]:
```suggestion
* Upgraded to Storybook 8.3.6.
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 7]
```diff
@@ -2,6 +2,79 @@
...
+
+New features
+------------
+
+* [#4320] Made it possible to let the backend drive the content of the confirmation page more.
+* [#4320] Moved cosign stories to shared location.
+* [#4320] Updated introduction text for cosigning.
+* [#4320] Support optional code for cosign login URL.
+* [#4320] Only display cosign login options if links in emails are not allowed.
+* [#4320] Set up dedicated route for cosign start.
```
[the remark this answers is not in the document]
Comment (vaszig) [reply CHANGELOG.rst:18]: Yes, this is what I meant on slack that I guess we do not need everything here.

File: CHANGELOG.rst (hunk 1 above)
Comment (vaszig) [reply CHANGELOG.rst:61]: Yes, I agree but I thought it's not the case for this specific verb since we use it a lot as 'ensure' in the changelog.

**Review by sergei-maertens — APPROVED** [review]
